### PR TITLE
Add support for bottlerocket kubernetes settings

### PR DESCRIPTION
**Pull request by abhinavmpandey08**
*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/5070

*Description of changes:*
Allow users to configure the following kubernetes settings on their EKS-A bottlerocket control plane and worker nodes
- allowed-unsafe-sysctls
-  max-pods
-  cluster-dns-ip

Example config
```yaml
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: VSphereMachineConfig
metadata:
  name: test
spec:
  datastore: "/SDDC-Datacenter/datastore/WorkloadDatastore"
  diskGiB: 22
  folder: "/SDDC-Datacenter/vm/eksa"
  memoryMiB: 8192
  numCPUs: 2
  osFamily: bottlerocket
  resourcePool: "/SDDC-Datacenter/host/Cluster-1/Resources/Compute-ResourcePool"
  hostOSConfiguration:
    bottlerocketConfiguration:
      kubernetes:
        maxPods: 50
        clusterDNSIPs:
          - 10.96.0.10
        allowedUnsafeSysctls:
          - "net.core.somaxconn"
          - "net.ipv4.ip_local_port_range"
  users:
  - name: ec2-user
```

Verified the config on the bottlerocket nodes using `apiclient`
```bash
$ apiclient get settings.kubernetes.max-pods
{
  "settings": {
    "kubernetes": {
      "max-pods": 50
    }
  }
}

$ apiclient get settings.kubernetes.allowed-unsafe-sysctls
{
  "settings": {
    "kubernetes": {
      "allowed-unsafe-sysctls": [
        "net.core.somaxconn",
        "net.ipv4.ip_local_port_range"
      ]
    }
  }
}

$ apiclient get settings.kubernetes.cluster-dns-ip
{
  "settings": {
    "kubernetes": {
      "cluster-dns-ip": "10.96.0.10"
    }
  }
}
```

*Testing (if applicable):*
Created the cluster and verified the kubernetes settings are being configured on bottlerocket propely

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: config/webhook/manifests.yaml [hunk 1]
```diff
@@ -34,9 +34,9 @@ webhooks:
     service:
```
Comment (abhinavmpandey08) [inline config/webhook/manifests.yaml:34]: This file is auto-generated. These changes originated from`make generate`

File: pkg/providers/vsphere/template.go [hunk 2]
```diff
@@ -295,8 +295,18 @@ func buildTemplateMapCP(
...
 	}
 
-	if controlPlaneMachineSpec.HostOSConfiguration != nil && controlPlaneMachineSpec.HostOSConfiguration.NTPConfiguration != nil {
-		values["cpNtpServers"] = controlPlaneMachineSpec.HostOSConfiguration.NTPConfiguration.Servers
+	if controlPlaneMachineSpec.HostOSConfiguration != nil {
+		if controlPlaneMachineSpec.HostOSConfiguration.NTPConfiguration != nil {
+			values["cpNtpServers"] = controlPlaneMachineSpec.HostOSConfiguration.NTPConfiguration.Servers
+		}
+
+		if controlPlaneMachineSpec.HostOSConfiguration.BottlerocketConfiguration != nil && controlPlaneMachineSpec.HostOSConfiguration.BottlerocketConfiguration.Kubernetes != nil {
```
Comment (ahreehong) [inline pkg/providers/vsphere/template.go:303]:
```suggestion

```
These if statements aren't needed

File: pkg/providers/vsphere/template.go [hunk 3]
```diff
@@ -398,8 +408,18 @@ func buildTemplateMapMD(
...
 	}
 
-	if workerNodeGroupMachineSpec.HostOSConfiguration != nil && workerNodeGroupMachineSpec.HostOSConfiguration.NTPConfiguration != nil {
-		values["ntpServers"] = workerNodeGroupMachineSpec.HostOSConfiguration.NTPConfiguration.Servers
+	if workerNodeGroupMachineSpec.HostOSConfiguration != nil {
+		if workerNodeGroupMachineSpec.HostOSConfiguration.NTPConfiguration != nil {
+			values["ntpServers"] = workerNodeGroupMachineSpec.HostOSConfiguration.NTPConfiguration.Servers
+		}
+
+		if workerNodeGroupMachineSpec.HostOSConfiguration.BottlerocketConfiguration != nil && workerNodeGroupMachineSpec.HostOSConfiguration.BottlerocketConfiguration.Kubernetes != nil {
```
Comment (ahreehong) [inline pkg/providers/vsphere/template.go:416]:
```suggestion

```
Same thing as above

File: pkg/providers/vsphere/template.go (hunk 2 above)
Comment (abhinavmpandey08) [reply pkg/providers/vsphere/template.go:303]: good point, lemme remove them

**Review by pokearu — APPROVED** [review]

**Pull request merged**